src/vs/workbench/services/positronDataExplorer/common/positronDataExplorerUri.ts

```diff
@@ -38,14 +38,22 @@ export class PositronDataExplorerUri {
 			return undefined;
 		}
 
-		// Parse the resource.
-		const match = resource.path.match(/^positron-data-explorer-([0-9a-fA-F]{8}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{12})$/);
-		const identifier = match?.[1];
-		if (typeof identifier !== 'string') {
+		// Parse the resource. Either it's a runtime comm id or a duckdb:$PATH identifier
+		const match = resource.path.match(
+			/^positron-data-explorer-(?:([0-9a-fA-F]{8}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{12})|(duckdb:.+))$/
+		);
```

Comment:
I hacked the URI for now to know what kind of backend (Jupyter comm or duckdb backend) is used. This could be generalized to support other kinds of non-comm-based backends in the future